sys/filedesc.c

```diff
@@ -87,9 +121,7 @@ fdtab_t *fdtab_copy(fdtab_t *fdt) {
 
   mtx_scoped_lock(&fdt->fdt_mtx);
 
-  /* We can assume both filedescs use 20 descriptors,
-   * because we don't support other numbers. */
-  assert(fdt->fdt_nfiles == newfdt->fdt_nfiles);
+  fd_growtable(newfdt, fdt->fdt_nfiles);
```

Comment:
`fd_growtable` does not work if `newsize` is smaller or equal to current size. Thus you should only call `growtable` here if it actually needs expanding.